File: library/src/test/java/com/proxerme/library/api/PageAdapterTest.java
package com.proxerme.library.api;

import com.proxerme.library.entitiy.manga.Page;
import com.squareup.moshi.JsonDataException;
import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatExceptionOfType;

/**
 * @author Ruben Gees
 */
public class PageAdapterTest {

    private PageAdapter adapter;

    @Before
    public void setUp() {
        adapter = new PageAdapter();
    }

    @Test
    public void testFromJson() {
        assertThat(adapter.fromJson(new String[][]{
                new String[]{"a", "1", "2"},
                new String[]{"b", "5", "3"}
        })).containsExactly(new Page("a", 1, 2), new Page("b", 5, 3));
    }

    @Test
    public void testFromJsonInvalidSize() {
        assertThatExceptionOfType(JsonDataException.class).isThrownBy(() -> adapter.fromJson(new String[][]{
                new String[]{"a", "1", "2"},
                new String[0]
        }));
    }

    @Test
    public void testFromJsonNoNumber() {
        assertThatExceptionOfType(NumberFormatException.class).isThrownBy(() -> adapter.fromJson(new String[][]{
                new String[]{"a", "invalid", "2"}
        }));
    }
}

File: library/src/test/kotlin/me/proxer/library/internal/adapter/PageAdapterTest.kt
package me.proxer.library.internal.adapter

import com.squareup.moshi.JsonDataException
import me.proxer.library.entity.manga.Page
import org.amshove.kluent.invoking
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldThrow
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class PageAdapterTest {

    private val adapter = PageAdapter()

    @Test
    fun testFromJson() {
        adapter.fromJson(arrayOf(arrayOf("a", "1", "2"), arrayOf("b", "5", "3"))) shouldBeEqualTo listOf(
            Page(name = "a", height = 1, width = 2),
            Page(name = "b", height = 5, width = 3)
        )
    }

    @Test
    fun testFromJsonInvalidSize() {
        invoking { adapter.fromJson(arrayOf(arrayOf("a", "1"))) } shouldThrow JsonDataException::class
    }

    @Test
    fun testFromJsonNoNumber() {
        invoking { adapter.fromJson(arrayOf(arrayOf("a", "invalid", "2"))) } shouldThrow JsonDataException::class
    }
}
